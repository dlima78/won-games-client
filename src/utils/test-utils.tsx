import { render, RenderOptions } from '@testing-library/react'
import {
  CartContext,
  CartContextData,
  CartContextDefaultValues
} from 'hooks/use-cart'
import {
  WishlistContext,
  WishlistContextData,
  WishlistDefaultValues
} from 'hooks/use-wishlist'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = {
  cartProviderProps?: CartContextData
  wishlistProviderProps?: WishlistContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  {
    cartProviderProps = CartContextDefaultValues,
    wishlistProviderProps = WishlistDefaultValues,
    ...renderOptions
  }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <CartContext.Provider value={cartProviderProps}>
        <WishlistContext.Provider value={wishlistProviderProps}>
          {ui}
        </WishlistContext.Provider>
      </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
