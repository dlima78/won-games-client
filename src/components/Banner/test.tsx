import { render, screen } from 'utils/test-utils'

import Banner from '.'

const props = {
  img: 'https://source.unsplash.com/user/willianjusten/1042x580',
  title: 'Death death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season</p>',
  buttonLabel: 'Buy Now',
  buttonLink: '/games/death-death'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    render(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /Death death/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /Death death/i })
    ).toBeInTheDocument()
  })

  it('should render a Ribbon', () => {
    render(
      <Banner
        {...props}
        ribbon="20% off"
        ribbonSize="small"
        ribbonColor="secondary"
      />
    )

    const ribbon = screen.getByText(/20% off/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({ backgroundColor: 'background: #3CD3C1' })
    expect(ribbon).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem'
    })
  })
})
