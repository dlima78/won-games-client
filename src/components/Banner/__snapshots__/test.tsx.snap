// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c0 {
  position: relative;
  box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
}

.c1 {
  background-color: #EAEAEA;
  background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);
  background-position: center center;
  background-size: cover;
  width: 100%;
  height: 23rem;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 400;
  margin-bottom: 2.4rem;
}

.c4 strong {
  color: #F231A5;
  font-weight: 600;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border: none;
  border-radius: 0.4rem;
  color: #FAFAFA;
  cursor: pointer;
  outline: none;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  height: 5rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

@media (min-width:768px) {
  .c1 {
    height: 50rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Death death"
    class="c1"
    role="image"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Death death
    </h2>
    <h3
      class="c4"
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="c5"
      href="/games/death-death"
    >
      <span>
        Buy Now
      </span>
    </a>
  </div>
</main>
`
